src/Components/AllTrips.js

```diff
@@ -0,0 +1,162 @@
+import { useEffect, useState } from "react";
+import { onValue, ref, getDatabase } from "firebase/database";
+import AnimatedPage from "./AnimatedPage";
+import firebase from "../firebase";
+
+const AllTrips = ({
+  trips,
+  setTrips,
+  countryData,
+  searchValue,
+  setSearchValue,
+}) => {
+  const [filteredTrips, setFilteredTrips] = useState([]);
+
+  const handleSearchChange = (e) => {
+    if (
+      searchValue.citySearch &&
+      searchValue.countrySearch &&
+      e.target.id === "countrySearch"
+    ) {
+      setSearchValue({
+        citySearch: "",
+        countrySearch: e.target.value,
+      });
+    } else {
+      setSearchValue({
+        ...searchValue,
+        [e.target.id]: e.target.value,
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (searchValue.citySearch && searchValue.countrySearch) {
+      const cityData = trips.filter((trip) => {
+        return trip.tripInfo.city.includes(searchValue.citySearch);
+      });
+
+      const newData = cityData.filter((trip) => {
+        return trip.tripInfo.country.includes(searchValue.countrySearch);
+      });
+
+      setFilteredTrips(newData);
```

Comment:
i feel like there could be a way to refactor this without setFilteredTrips(newData) for each else if?

Reply:
re-factored! i took the declaration of newData out of the if statement and put it in the function scope as `let newData`. now they're re-assigned in each conditional, and i was able to replace all of the `setFilteredTrips(newData)` calls with one call at the end of the function 😎